src/main/java/io/vertx/core/http/impl/HttpClientImpl.java

```diff
@@ -138,6 +139,8 @@ public HttpClientImpl(VertxInternal vertx, HttpClientOptions options, CloseFutur
     this.metrics = vertx.metricsSPI() != null ? vertx.metricsSPI().createHttpClientMetrics(options) : null;
     this.options = new HttpClientOptions(options);
     this.closeFuture = closeFuture;
+    ContextInternal ctx = vertx.getOrCreateContext();
```

Comment:
no need to get or create context, simply call `vertx.createEventLoopContext()`

Reply:
done